src/solvers/dgsem_tree/dg_2d_subcell_limiters.jl

```diff
@@ -155,38 +190,271 @@ end
+
+    # The FV-form fluxes are calculated in a recursive manner, i.e.:
+    # fhat_(0,1)   = w_0 * FVol_0,
+    # fhat_(j,j+1) = fhat_(j-1,j) + w_j * FVol_j,   for j=1,...,N-1,
+    # with the split form volume fluxes FVol_j = -2 * sum_i=0^N D_ji f*_(j,i).
+
+    # To use the symmetry of the `volume_flux`, the split form volume flux is precalculated
+    # like in `calc_volume_integral!` for the `VolumeIntegralFluxDifferencing`
+    # and saved in in `flux_temp`.
+
+    # Split form volume flux in orientation 1: x direction
+    flux_temp .= zero(eltype(flux_temp))
+    flux_temp_noncons .= zero(eltype(flux_temp_noncons))
+
+    for j in eachnode(dg), i in eachnode(dg)
+        u_node = get_node_vars(u, equations, dg, i, j, element)
+
+        # All diagonal entries of `derivative_split` are zero. Thus, we can skip
+        # the computation of the diagonal terms. In addition, we use the symmetry
+        # of `volume_flux_cons` and `volume_flux_noncons` to save half of the possible two-point flux
+        # computations.
+        for ii in (i + 1):nnodes(dg)
+            u_node_ii = get_node_vars(u, equations, dg, ii, j, element)
+            flux1 = volume_flux_cons(u_node, u_node_ii, 1, equations)
+            multiply_add_to_node_vars!(flux_temp, derivative_split[i, ii], flux1,
+                                       equations, dg, i, j)
+            multiply_add_to_node_vars!(flux_temp, derivative_split[ii, i], flux1,
+                                       equations, dg, ii, j)
+            for noncons in 1:nnoncons(equations)
+                # We multiply by 0.5 because that is done in other parts of Trixi
+                flux1_noncons = 0.5 *
+                                volume_flux_noncons(u_node, u_node_ii, 1, equations,
+                                                    NonConservativeSymmetric(), noncons)
+                multiply_add_to_node_vars!(flux_temp_noncons, derivative_split[i, ii],
+                                           flux1_noncons,
+                                           equations, dg, noncons, i, j)
+                multiply_add_to_node_vars!(flux_temp_noncons, derivative_split[ii, i],
+                                           flux1_noncons,
+                                           equations, dg, noncons, ii, j)
+            end
+        end
+    end
 
-# Calculate the antidiffusive flux `antidiffusive_flux` as the subtraction between `fhat` and `fstar`.
-@inline function calcflux_antidiffusive!(fhat1, fhat2, fstar1, fstar2, u, mesh,
-                                         nonconservative_terms, equations,
+    # FV-form flux `fhat` in x direction
+    fhat1_L[:, 1, :] .= zero(eltype(fhat1_L))
+    fhat1_L[:, nnodes(dg) + 1, :] .= zero(eltype(fhat1_L))
+    fhat1_R[:, 1, :] .= zero(eltype(fhat1_R))
+    fhat1_R[:, nnodes(dg) + 1, :] .= zero(eltype(fhat1_R))
+
+    fhat_temp[:, 1, :] .= zero(eltype(fhat1_L))
+    fhat_noncons_temp[:, :, 1, :] .= zero(eltype(fhat1_L))
```

Comment:
```suggestion
    for j in axes(fhat_noncons_temp, 4)
        fhat_noncons_temp[:, :, 1, j] .= zero(eltype(fhat1_L))
    end
```
In the past, I sometimes had issues with pointwise overwriting arrays of higher dimension.
While any version doesn't have allocations, the second one is in fact slightly faster.
```
julia> @btime Trixi.calcflux_fhat!($fhat1_L, $fhat1_R, $fhat2_L, $fhat2_R, $u, $mesh,
                              $nonconservative_terms, $equations, $volume_flux_dg, $solver, $1,
                              $cache)
  2.182 μs (0 allocations: 0 bytes)
```

old version:
```
julia> @btime Trixi.calcflux_fhat!($fhat1_L, $fhat1_R, $fhat2_L, $fhat2_R, $u, $mesh,
                              $nonconservative_terms, $equations, $volume_flux_dg, $solver, $1,
                              $cache)
  2.202 μs (0 allocations: 0 bytes)
```
Is that something that happens more often or is it just a bit random here (or maybe an analysis error from me)? @ranocha 
Replacing the other reset in line 318 does not change anything.

Reply:
Thanks!
I tried using loops for all indices, but the performance does not improve significantly :frowning_face: